### ISSUE TITLE
cluster-api security self assessment initiative 

### ISSUE THREAD

**Issue by rficcaglia** [kubernetes/community#5814]
This is to consolidate info and inform all the interested folks regarding discussions around security assessment of cluster-api sub-project.

This is our first attempt as a community where a Security SIG will perform a security assessment of a sub-project of a graduated CNCF project. So we expect to make slow but reasonable progress and will be open for feedback on how we can improve for the benefit of others who will follow us.

The first step of a (community driven) security assessment of a project is a self-assessment of the project. Although, this is typical done by the project maintainers, some of us in #sig-security  would be happy to take a first crack at filling out the outline template provided by CNCF TAG-Security.

Currently the volunteers are educating themselves on cluster-api docs and presentations and will coordinate with maintainers (who understandably are very busy but are engaged).

Related items:

https://github.com/kubernetes/community/pull/5792

https://github.com/cncf/tag-security/issues/603

https://groups.google.com/g/kubernetes-sig-cluster-lifecycle/c/Fi0UGzfbQfY

**Comment by rficcaglia** [comment]
/sig sig-security
/sig cluster-api

maybe we should/need to create a wg? @tabbysable @reylejano @PushkarJ ?

**Comment by reylejano** [comment]
Related issue in cluster-api repo https://github.com/kubernetes-sigs/cluster-api/issues/4446

**Comment by neolit123** [comment]
> maybe we should/need to create a wg?

WGs have a contract with Kubernetes Steering and require a community page, Zoom call slots, Slack channels and yearly status reports. my vote would be to not create WGs for subproject security audits, but if people insist it can be done.

**Comment by PushkarJ** [comment]
@rficcaglia regarding WG creation, our co-chair @tabbysable PoV is [here](https://github.com/kubernetes/community/pull/5792#issuecomment-846483164), which I tend to agree with. 

TL;DR this seems like a lower intensity effort that would not require a WG

**Comment by tabbysable** [comment]
For the record: this is wonderful, huge thanks to everyone involved.

**Comment by rficcaglia** [comment]
here's my first pass at the self-assessment outline structure, ie not the CAPI details themselves but the high level parts to be filled in (including a place for someone to fill in the CAPI-specific features and controls). this is meant to be both for CAPI and serve as a template for future subproject use.

https://docs.google.com/document/d/1Fj_cLUN9kLruHbEgmYiEgoqZjf2rRuVOmQDGOKByaf4/edit?usp=sharing
